provider/src/main/java/com/cloudera/director/google/sql/GoogleCloudSQLProvider.java

```diff
@@ -0,0 +1,557 @@
+          Operation subjectOperation = sqlAdmin.operations().get(projectId, pendingOperationName).execute();
+          OperationErrors errors = subjectOperation.getError();
+          boolean isActualError = false;
+
+          if (errors != null) {
+            List<OperationError> errorsList = errors.getErrors();
+
+            if (errorsList != null) {
+              for (OperationError error : errorsList) {
+                String operationType = subjectOperation.getOperationType();
+                String errorCode = error.getCode();
+
+                // We want insertion and deletion operations to be idempotent.
+                if (operationType.equals("insert") && errorCode.equals("RESOURCE_ALREADY_EXISTS")) {
+                  LOG.info("Attempted to create resource '{}', but it already exists.",
+                      Urls.getLocalName(subjectOperation.getTargetLink()));
+                } else {
+                  accumulator.addError(null, error.getMessage());
+                  isActualError = true;
+                }
+              }
+            }
+          }
+
+          if (acceptableStates.contains(subjectOperation.getStatus())) {
+              completedOperations.add(pendingOperation);
+
+            if (!isActualError) {
+              successfulOperations.add(pendingOperation);
+            }
+          }
+        } catch (IOException e) {
+          accumulator.addError(null, e.getMessage());
+        }
+      }
+
+      // Remove all operations that reached an acceptable state.
+      pendingOperations.removeAll(completedOperations);
+
+      if (pendingOperations.size() > 0 && totalTimePollingSeconds > pollingTimeoutSeconds) {
+        List<String> pendingOperationNames = new ArrayList<String>();
+
+        for (Operation pendingOperation : pendingOperations) {
+          pendingOperationNames.add(pendingOperation.getName());
+        }
+
+        accumulator.addError(null, "Exceeded timeout of '" + pollingTimeoutSeconds + "' seconds while " +
+            "polling for pending operations to complete: " + pendingOperationNames);
+
+        timeoutExceeded = true;
+      } else {
+        // Update polling interval.
+        int oldIncrement = pollIncrement;
+        pollIncrement = pollInterval;
+        pollInterval += oldIncrement;
+        pollInterval = Math.min(pollInterval, maxPollingIntervalSeconds);
+      }
+    }
+
+    return successfulOperations;
+  }
+
+  private boolean prefixIsValid(GoogleCloudSQLInstanceTemplate template,
+      LocalizationContext templateLocalizationContext) {
+    PluginExceptionConditionAccumulator accumulator = new PluginExceptionConditionAccumulator();
+
+    GoogleCloudSQLInstanceTemplateConfigurationValidator.checkPrefix(template, accumulator, templateLocalizationContext);
+
+    boolean isValid = accumulator.getConditionsByKey().isEmpty();
+
+    if (!isValid) {
+      LOG.info("Instance name prefix '{}' is invalid.",
```

Comment:
Why is a method such as `prefixIsValid()` also logging rather than just returning a boolean and letting the caller decide whether or not to log this issue?